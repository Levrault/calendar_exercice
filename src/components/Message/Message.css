.container {
  align-items: center;
  display: flex;
  justify-content: center;
  
  &.success {
    color: $palette-success;
  }

  &.error {
    color: $palette-error;
  }
}

.icon {
  margin-right: $space-s;
}
