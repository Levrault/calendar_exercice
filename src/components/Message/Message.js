import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './Message.css';

/**
 * Message
 * @param {string} type
 * @returns {node}
 */
const Message = ({ type, children }) => {
  const hasSuccess = type === 'success';
  const hasError = type === 'error';
  const containerClassname = classnames(styles.container, {
    [styles.success]: hasSuccess,
    [styles.error]: hasError
  });
  const iconClassname = classnames('material-icons', styles.icon);
  return (
    <div className={containerClassname}>
      {hasSuccess && <i className={iconClassname}>check_circle</i> }
      {hasError && <i className={iconClassname}>error</i> }
      {children}
    </div>
  );
};

Message.propTypes = {
  type: PropTypes.string
};

Message.defaultProps = {
  type: ''
};

export default Message;
