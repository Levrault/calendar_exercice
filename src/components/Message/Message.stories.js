import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs/react';
import Wrapper from '../../tools/storybook/Wrapper';
import Message from './Message';

storiesOf('Message', module)
  .addDecorator(withKnobs)
  .addDecorator(Wrapper)
  .add('default', () => (
    <Message type={text('type', '')}>Message</Message>
  ))
  .add('with type "success"', () => (
    <Message type={text('type', 'success')}>Message</Message>
  ))
  .add('with type "error"', () => (
    <Message type={text('type', 'error')}>Message</Message>
  ));
