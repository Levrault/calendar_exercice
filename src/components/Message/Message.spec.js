import React from 'react';
import renderer from 'react-test-renderer';
import Message from './Message';

describe('Message component', () => {
  it('should render with empty type', () => {
    const component = renderer.create(
      <Message>Message</Message>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render when type has "success" value', () => {
    const component = renderer.create(
      <Message type="success">Message</Message>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render when type has "error" value', () => {
    const component = renderer.create(
      <Message type="error">Message</Message>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
