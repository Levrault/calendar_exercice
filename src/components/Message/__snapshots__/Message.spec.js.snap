// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render when type has "error" value 1`] = `
<div
  className="undefined"
>
  <i
    className="material-icons"
  >
    error
  </i>
  Message
</div>
`;

exports[`Message component should render when type has "success" value 1`] = `
<div
  className=""
>
  <i
    className="material-icons"
  >
    check_circle
  </i>
  Message
</div>
`;

exports[`Message component should render with empty type 1`] = `
<div
  className=""
>
  Message
</div>
`;
